shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/data/remote/apiclient/core/ApiClient.kt

```diff
@@ -0,0 +1,110 @@
+package co.nimblehq.ic.kmm.suv.data.remote.apiclient.core
+
+import co.nimblehq.ic.kmm.suv.BuildKonfig
+import co.nimblehq.ic.kmm.suv.data.local.datasource.TokenLocalDataSource
+import co.nimblehq.ic.kmm.suv.data.remote.body.RefreshTokenApiBody
+import co.nimblehq.ic.kmm.suv.data.remote.apiclient.builder.path
+import co.nimblehq.ic.kmm.suv.data.remote.model.TokenApiModel
+import co.nimblehq.ic.kmm.suv.domain.model.AppError
+import co.nimblehq.jsonapi.json.JsonApi
+import co.nimblehq.jsonapi.model.JsonApiException
+import io.github.aakira.napier.DebugAntilog
+import io.github.aakira.napier.LogLevel
+import io.github.aakira.napier.Napier
+import io.ktor.client.*
+import io.ktor.client.engine.*
+import io.ktor.client.plugins.auth.*
+import io.ktor.client.plugins.auth.providers.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.plugins.logging.*
+import io.ktor.client.plugins.logging.LogLevel.ALL
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import io.ktor.serialization.kotlinx.json.*
+import kotlinx.coroutines.flow.*
+import kotlinx.serialization.json.Json
+
+class ApiClient(engine: HttpClientEngine, tokenLocalDataSource: TokenLocalDataSource? = null) {
+
+    val httpClient: HttpClient
+    val json = Json {
+        prettyPrint = true
+        isLenient = true
+        encodeDefaults = true
+        ignoreUnknownKeys = true
+    }
+
+    private val tokenLocalDataSource: TokenLocalDataSource?
+
+    init {
+        this.tokenLocalDataSource = tokenLocalDataSource
+        Napier.takeLogarithm()
+        Napier.base(DebugAntilog())
+        httpClient = HttpClient(engine = engine) {
+            install(Logging) {
+                level = ALL
+                logger = object : Logger {
+                    override fun log(message: String) {
+                        Napier.log(LogLevel.DEBUG, message = message)
+                    }
+                }
+            }
+
+            install(ContentNegotiation) {
+                json(json)
+            }
+
+            tokenLocalDataSource?.let {
+                install(Auth) {
+                    bearer {
+                        loadTokens {
+                            val token = it.getToken()
+                            token?.let {
+                                BearerTokens(it.accessToken, it.refreshToken)
+                            }
```

Comment:
too many vague `it` usage

```suggestion
                            tokenLocalDataSource.getToken()?.run {
                                BearerTokens(accessToken, refreshToken)
                            }
```

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/94/commits/fca5d8acb0b263ec50661cd79bd5560e4cad0f8f